docs/src/advanced-concepts/auxiliary-outputs.rst

```diff
@@ -0,0 +1,65 @@
+Auxiliary outputs
+=================
+
+These outputs, which are idenfified by the ``mtm::aux::`` prefix,
+represent additional information that the model may provide. They are not
+conventional trainable outputs, and they often correspond to internal
+information that the model is capable of providing, such as its internal
+representation.
+
+The following auxiliary outputs that are currently supported
+by one or more architectures in the library:
+
+- ``mtm::aux::last_layer_features``: The internal representation
+   of the model at the last layer, before the final linear transformation.
+
+The following table shows the architectures that support each of the
+auxiliary outputs:
+
++------------------------------------------+-----------+------------------+-----+
+| Auxiliary output                         | SOAP-BPNN | Alchemical Model | PET |
++------------------------------------------+-----------+------------------+-----+
+| ``mtm::aux::last_layer_features``        | Yes       |       No         | No  |
++------------------------------------------+-----------+------------------+-----+
```

Comment:
Are we sure we want this table here and not do it the other way around and put a table of supported auxiliary outputs on the page of each model?

I am afraid that we just forget to update this table...

Reply:
I agree that it would be easier for us, but for the user I think it's better to have it in a centralized place, where they can think "well, I want this, I can get it with this architecture, but not this other one"

Reply:
Long term we could try to auto-generate these tables from the code, and have it in both places. For now either place is fine with me!